addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/internal/core/LxWsClient.java

```diff
@@ -644,141 +668,295 @@ public void onMessage(String msg) {
+        LxJsonSubResponse sendCmdWithResp(String command, boolean sync, boolean encrypt) {
+            responseLock.lock();
+            try {
+                if (commandResponse != null) {
+                    logger.warn("[{}] Command not sent, previous command not finished: {}", debugId, command);
+                    return null;
+                }
+                if (!sendCmdNoResp(command, encrypt)) {
+                    return null;
+                }
+                commandResponse = new LxJsonSubResponse();
+                commandResponse.control = command;
+                LxJsonSubResponse response = commandResponse;
+                syncRequest = sync;
+                if (sync) {
+                    if (!responseAvailable.await(connectTimeout, TimeUnit.SECONDS)) {
+                        commandResponse = null;
+                        responseTimeout();
+                        return null;
+                    }
+                    commandResponse = null;
+                }
+                return response;
+            } catch (InterruptedException e) {
+                logger.debug("[{}] Interrupted waiting for response: {}", debugId, command);
+                commandResponse = null;
+                return null;
+            } finally {
+                responseLock.unlock();
             }
         }
 
         /**
-         * Sends a string command to the Miniserver
+         * Sends a command to the Miniserver and encrypts it if command can be encrypted and encryption is available.
+         * The request is asynchronous and no response is expected. It can be used to send commands from the websocket
+         * thread or commands for which the responses are not following the standard format defined in
+         * {@link LxJsonSubResponse}.
+         * If the caller expects the non-standard response it should manage its reception and the response timeout.
          *
-         * @param string
+         * @param command
          *            command to send to the Miniserver
-         * @throws IOException
-         *             exception when communication error occurs
+         * @param encrypt
+         *            true if command can be encrypted
+         * @return
+         *         true if command was sent (no information if it was received)
          */
-        private void sendString(String string) throws IOException {
+        private boolean sendCmdNoResp(String command, boolean encrypt) {
             synchronized (state) {
                 if (session != null && state != ClientState.IDLE && state != ClientState.CONNECTING
                         && state != ClientState.CLOSING) {
-                    logger.debug("[{}] sending command: {}", debugId, string);
-                    session.getRemote().sendString(string);
+                    String encrypted = encrypt ? security.encrypt(command) : command;
+                    if (logger.isDebugEnabled() && encrypted.equals(command)) {
```

Comment:
Your probably do the debug enabled check to avoid the equals? But it looks you could also do `!encrypt`? (Or better do a positive condition check and reverse the logging statements.)

Reply:
Event with encrypt == true, the encryption implementing class may not encrypt and return the original string. Yes, I check the debug flag to avoid the string comparison. I changed it a bit to better show this logic.